azure-iot-device/azure/iot/device/common/alarm.py

```diff
@@ -25,10 +32,13 @@ def cancel(self):
     def run(self):
         """Method representing the thread's activity.
         Overrides the method inherited from Thread.
-        Will invoke the Alarm's given function at the given alarm time.
+        Will invoke the Alarm's given function at the given alarm time (accurate within 1 second)
         """
-        interval = self.alarm_time - time.time()
-        self.finished.wait(interval)
+        curr_time = time.time()
```

Comment:
Not sure of the utilty of the curr_time variable.

Reply:
Good point.